collect_app/src/main/java/org/odk/collect/android/formentry/FormSessionRepository.kt

```diff
@@ -37,6 +39,12 @@ class AppStateFormSessionRepository(application: Application) : FormSessionRepos
         getLiveData(id).value = FormSession(formController, form, instance)
     }
 
+    override fun update(id: String, instance: Instance?) {
+        getLiveData(id).value?.let {
+            getLiveData(id).value = FormSession(it.formController, it.form, instance)
```

Comment:
Could this just use `copy`? You also shouldn't need to do the live data look up (`getLiveData` twice here).

Same for the `InMemFormSessionRepository`

Reply:
Done.